syntax = "proto3";

message AppearanceState {
  string name = 1;
  string prefab_name = 2;
}

message AppearanceTransition {
  string trigger_name = 1;
  string prev_state_name = 2;
  string next_state_name = 3;
  string prefab_name = 4;
}

message AppearanceStateMachine {
  string name = 1;
  repeated AppearanceState states = 2;
  string initial_state_name = 3;
  repeated AppearanceTransition transitions = 4;
}

message AnimationAssetBase {
  repeated AppearanceStateMachine appearance_state_machine = 1;
}